Subject: [PATCH] build: Unset JAVA_TOOL_OPTIONS to read java version

If env variable JAVA_TOOL_OPTIONS is set, it breaks the expected java
version output. Exp:

 Picked up JAVA_TOOL_OPTIONS: -javaagent:/usr/share/java/jayatanaag.jar
 java version "1.7.0_79"
 OpenJDK Runtime Environment (IcedTea 2.5.5) (7u79-2.5.5-0ubuntu1)
 OpenJDK 64-Bit Server VM (build 24.79-b02, mixed mode)

Unset JAVA_TOOL_OPTIONS for the purposes of reading java and javac
versions. This was encountered on an Ubuntu 15.04 system.

diff --git a/core/main.mk b/core/main.mk
index 8643ee3..d816ff4 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -188,8 +188,8 @@
 endif
 
 ifeq ($(JAVA_NOT_REQUIRED), false)
-java_version_str := $(shell unset _JAVA_OPTIONS && java -version 2>&1)
-javac_version_str := $(shell unset _JAVA_OPTIONS && javac -version 2>&1)
+java_version_str := $(shell unset _JAVA_OPTIONS JAVA_TOOL_OPTIONS && java -version 2>&1)
+javac_version_str := $(shell unset _JAVA_OPTIONS JAVA_TOOL_OPTIONS && javac -version 2>&1)
 
 # Check for the correct version of java, should be 1.8 by
 # default and only 1.7 if LEGACY_USE_JAVA7 is set.
